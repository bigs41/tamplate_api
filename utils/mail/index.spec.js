const { mailer, sendMailWithTemplate } = require('../mail')

describe('utils', () => {
  describe('mail', () => {
    it('should send mail with template', async done => {
      mailer.sendMail = jest.fn()
      await sendMailWithTemplate('vertifyOtp', 'user@mail.com', 'รีเซ็ตรหัสผ่าน',
        {
          username: 'ชื่อ',
          vertifyOtp: '123456',
          origin: 'https://[::1]'
        }
      )

      expect(mailer.sendMail).toMatchSnapshot()
    })
  })
})