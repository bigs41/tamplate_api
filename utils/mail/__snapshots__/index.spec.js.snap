// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils mail should send mail with template 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "html": <html lang="en"
      data-server-rendered="true"
>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <meta http-equiv="X-UA-Compatible"
          content="ie=edge"
    >
    <title>
      Document
    </title>
  </head>
  <body>
    <img src="https://[::1]/mail/banner.png"
         width="450"
         height="107"
    >
    <pre>
      บัญชีผู้ใช้ : ชื่อ
    OTP : 123456
    </pre>
  </body>
</html>,
        "subject": "รีเซ็ตรหัสผ่าน",
        "to": "user@mail.com",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
